calculator.js

```diff
@@ -0,0 +1,47 @@
+const operators = ['+', '-', '*', '/'];
+
+exports.create_calculator = function (operator) {
+  const numbers = [];
+
+  return {
+    add_number: function (number) {
+      numbers.push(number);
+    },
+    calculate: function () {
+      // Return zero if there are no numbers to calculate
+      if (numbers.length === 0) {
+        throw 'Cannot calculate a result because no numbers were given';
+      }
+
+      let answer = numbers[0];
+
+      let i = 0;
```

Comment:
Using i in a forEach is a bit confusing here - i is conventional for an if statement... You might wig out some of the nerds

Reply:
What would you recommend? Can "i" just be removed from this code? Would it change it's behaviour? How could this code be rewritten to avoid "i"?

Reply:
(note - my vs code debugger is broken and I can't seem to work out what is wrong)

i certainly does something, as when I remove all references, the answer to 2+4 = 8. I can see that i prevents an 'additional forEach' being carried out, as the forEach takes the answer on the first loop to be numbers[0] + numbers [0] (answer = answer + number). 

Need to now work out how to stop that happening. My instinct is that starting out by making answer in this way (combining it with number) feels wrong.

Reply:
So you have the jist of the problem identified. Were it not for "i" we would add two copies of the first number. 

I think setting answer to initially be the first number in the list is fine, but it just means we need a way to skip the first number when we enter the loop. And the easiest way to do that would be replace the number.forEach with a for(let i = 1... style loop. Note that I put **i = 1** to skip the first item.